upnp_port_forward/client.py

```diff
@@ -96,16 +108,14 @@ def _find_internal_ip_on_device_network(upnp_dev: upnpclient.upnp.Device) -> str
     raise _NoInternalAddressMatchesDevice(parsed_url.hostname)
 
 
-WAN_SERVICE_NAMES = (
-    "WANIPConn1",
-    "WANIPConnection.1",  # Nighthawk C7800
-    "WANPPPConnection.1",  # CenturyLink C1100Z
-    "WANPPPConn1",  # Huawei B528s-23a
-)
-
+def _get_wan_service(
+    upnp_dev: upnpclient.upnp.Device, required_service_names: Optional[Tuple[str, ...]]
+) -> upnpclient.upnp.Service:
 
-def _get_wan_service(upnp_dev: upnpclient.upnp.Device) -> upnpclient.upnp.Service:
-    for service_name in WAN_SERVICE_NAMES:
+    service_names_on_trial = (
```

Comment:
Another name that might be a bit more intuitive than `trial` might be `candidate_service_names`